### PR TITLE
Implement Complaints Functionality and Popup Reporting Feature

### PR DESCRIPTION
**Pull request by beviggiano**
This PR introduces a complete implementation of the complaints entity and integrates a popup reporting feature for posts.

### CODE REVIEW

File: backend/src/infra/api/modules/complaints/controller.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { Request, Response, NextFunction } from "express";
+import { createComplaintFactory, deleteComplaintFactory, getAllComplaintsFactory, getSpecificComplaintFactory, updateComplaintFactory } from "@src/modules/complaints/factory";
+import { InvalidParamError } from "../../errors/InvalidParamError";
+import { QueryError } from "../../errors/QueryError";
+
+class ComplaintsController {
```
Comment (edu-correia) [inline backend/src/infra/api/modules/complaints/controller.ts:6]: Missing docs comments for this controller

File: backend/src/infra/api/modules/complaints/routes.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ *         reason: "Inappropriate content"
+ *         createdAt: "2023-06-15T00:00:00Z"
+ */
+
+const router = Router();
+
+router.get("/all", verifyJWT, complaintsController.getAll);
+router.get("/:id", verifyJWT, complaintsController.getOne);
+router.post("/", verifyJWT, complaintsController.create);
+router.put("/:id", verifyJWT, complaintsController.update); // Adicionado
```
Comment (edu-correia) [inline backend/src/infra/api/modules/complaints/routes.ts:51]: We can remove this "// Adicionado" comment

File: frontend/src/hooks/usePostComplaints.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  const handleClosePopup = () => {
+    setShowPopup(false);
+    setReason('');
+  };
+
+  const handleSubmit = async () => {
+    try {
+      await axios.post('http://localhost:3333/complaints', {
```
Comment (edu-correia) [inline frontend/src/hooks/usePostComplaints.ts:19]: Do not use the api like this, use like this:

```ts
import api from "<path>/src/api/api.ts";

await api.post('/complaints') ...
```
This way, we have one file that manages our **backend url** for the entire project

**Review by edu-correia — CHANGES_REQUESTED** [review]
3 points to give a look

File: frontend/src/api/api.ts
Comment (edu-correia) [inline frontend/src/api/api.ts:1]: There is already a file just like this in src/services/api.ts
Please delete this one and update the import on the other files
Sorry if my last comment was misleading on where the file was located.

**Review by edu-correia — CHANGES_REQUESTED** [review]
one small change

**Review by juancoelhoo — APPROVED** [review]

**Review by edu-correia — APPROVED** [review]
LGTM

**Pull request merged**